### ISSUE TITLE
Need to bring (deprecated?) application.conf file

### ISSUE THREAD

**Issue by mfeblowitz** [Andarius/nginx-upload#7]
As documented in #4 and in the README update, I am using the `-v` directive to bring in the nginx.conf file and another `-v` directive to bring in a legacy `application.conf`, formerly residing in a sites-enabled dir. The config is received, but it appears that some of the directives are no longer recognized. E.g., `upload_buffer_size 10M;`, `upload_pass  /extract/;`, and `upload_store /tmp/upload 1;`. I suspect there are others. 

Has the protocol changed to deprecate these, or could it be that the latest `nginx.conf` needs to be paired with application.conf?

**Comment by Andarius** [comment]
I'm not sure. Can you provide a full repro for me to test?

**Comment by mfeblowitz** [comment]
Too many not yet open source files to provide for a test. I believe you can at least test run the attached config file. 

I was able to get at least part of this somewhat complex setup to work. I'm using a docker network to connect the nginx-upload container to the application container.  The simple pass-through parts of the configuration work:
```
  location /alive {
       resolver 127.0.0.11 ipv6=off;
       proxy_socket_keepalive on;
       set $cmc cefas-main-container:5001;
       proxy_pass  http://$cmc$request_uri;
       proxy_read_timeout 600s;
       rewrite_log on;
       }
```
This passes the `alive` request through to the `cefas-main-container.

However, the one part that is supposed to use the upload capability doesn't load without errors:
```
    listen        5000;
    server_name   cefas;
    client_max_body_size 500M;
    upload_buffer_size 10M;
    proxy_request_buffering off;
    error_log logs/error.log;
    error_log logs/error.log notice;
    error_log logs/error.log info;
    proxy_buffering off;
location /extract/ {
        # After upload, pass altered request body to this REST endpoint
        upload_pass  /extract/;
        # Store files to this directory, should exist as well as subdirs 0, 1....9
        upload_store /tmp/upload 1;
        upload_store_access user:r;
        # Set specified fields in request body
        upload_set_form_field $upload_field_name.name "$upload_file_name";
        upload_set_form_field $upload_field_name.content_type "$upload_content_type";
        upload_set_form_field $upload_field_name.path "$upload_tmp_path";
        # Inform backend about hash and size of a file
        upload_aggregate_form_field "$upload_field_name.md5" "$upload_file_md5";
        upload_aggregate_form_field "$upload_field_name.size" "$upload_file_size";
        # Here is where you define additional fields to pass through
        upload_pass_form_field "^drivers$";
        # set this to the error return codes that will cause the uploaded files to be deleted
        upload_cleanup 400 404 499 500-505;
        access_log /dev/stdout;
        error_log /dev/stderr;
       }
```
The items not loading correctly are:
```
    upload_buffer_size 10M;
    ...
    location /extract/ {
        ....
        upload_pass  /extract/;
        upload_store /tmp/upload 1;
        upload_store_access user:r;
        upload_set_form_field $upload_field_name.name "$upload_file_name";
        upload_set_form_field $upload_field_name.content_type "$upload_content_type";
        upload_set_form_field $upload_field_name.path "$upload_tmp_path";
        upload_aggregate_form_field "$upload_field_name.md5" "$upload_file_md5";
        upload_aggregate_form_field "$upload_field_name.size" "$upload_file_size";
        upload_pass_form_field "^drivers$";
        upload_cleanup 400 404 499 500-505;
        ...
```
The typical errors on run are, e.g.:
```
   [emerg] 1#1: unknown directive "upload_buffer_size" in /etc/nginx/conf.d/default.conf:5.
```
and
```
   [emerg] unknown directive "upload_pass" in /etc/nginx/conf.d/default.conf:13
```
An so forth.

My run command looks like this:
```
docker run --rm   -p 5000:5000 --network cefas-nginx-network -v  "$(pwd)/cefas-nginx-docker.conf:/etc/nginx/conf.d/default.conf:ro" --name nginx-upload nginx-upload:dev
```
Note that only one volume is mounted. I have tried with a mount for `nginx.conf` and that made no difference.

Perhaps there's some configuration missing that activates the unrecognized directives?

[cefas-nginx-docker.conf.txt](https://github.com/Andarius/nginx-upload/files/10548175/cefas-nginx-docker.conf.txt)

**Comment by Andarius** [comment]
Have you tried adding `load_module modules/ngx_http_upload_module.so;` ?

**Comment by mfeblowitz** [comment]
Wait - I got it all to work! 

I needed to grab some parts from your nginx.conf and include them in my `cefas-nginx-docker.conf` and then mount that as `/etc/nginx/nginx.conf`.

And that is what you just suggested.

I also needed to surround my file with `http { ... }`. Not sure whether it's needed, but I also included the empty `events { }` section.

( The load_module declaration alone didn't do it.) And in had to override the default nginx.conf - putting it in /etc/nginx/conf.d wasn't enough.)
